### DISCUSSION TITLE
Waveform viewer

### DISCUSSION THREAD

**Discussion by medwatt** [ra3xdh/qucs_s#352] [Ideas]
If I may ask, what is the rational behind not having a waveform viewer that allows zooming in, having cursors, etc., the sort of thing that almost all spice simulator GUIs have?

**Comment by ra3xdh** [comment]
The plotting susbsystem of Qucs-S was designed around 2003 for the Qt3 version of the application. I am only maintaining this code. I cannot say why the developers prefer to place diagram on DPL page or schematic instead of having waveform viewer tool. I suspect the concept of Qucs was copied from some early version of Agilent ADS software. 

Yes, in principle I am agreed that adding waveform viewer may make the application more user-friendly. But this may require the deep redesign of the existing code. I am developing the Qucs-S in my spare time and cannot perform this task by myself. The project may fall into the eternal refactoring if no full time developers will be assigned for such tasks. Currently the Qucs-S project has no full-time developers like KiCAD that is developed by CERN.  So, don't expect the redesign of the application in the near future.

**Reply by medwatt** [reply]
Thanks for your response. While I think the current way of displaying plots looks "cool" in screenshots, it has very limited usefulness.

**Reply by 2nOrderEDO** [reply]
Hello,

Sorry for replying to a closed topic. I find that both ways of visualization are useful. Plots on sheet are nice for documentation, while the waveform viewer is better in for simulation understanding and exploration.

Many people is used to the way LTSpice works, and I believe having a similar waveform viewer could draw interest to QUCS-S. I have a feeling that ngspice is as powerful (or even more) than QSPICE, specially since d_process was merged. The limiting factor for more widespread usage is on providing seamless access to such features. I see QUCS-S could fulfill this.

Awesome work so far, QUCS-S is developing very rapidly.

Regards

**Reply by ra3xdh** [reply]
@2nOrderEDO Yes, I am agreed that the "everything is component" concept inspired by ADS software was not a good idea. The traditional waveform viewer would be better. But the adding the waveform viewer in the existing design is not a straightforward task, but it is technically possible. There is no full time developers in the Qucs-S project and it is not supported by any organization. And I don't consider such massive new features addition may be implemented by myself. The only way you can speed up the adding the new features like waveform viewer is to provide a patch yourself or find a developer that will implement it. Fro example the working on the open-source project may be good thesis task for BS or MS student.  Making a donation also speeds up a development. 

> specially since d_process was merged

The adding of `d_process` and `d_cosim` in the Qucs-S GUI is planned, but I cannot say when the works on this feature will be started. May be this will appear in v25.1.0

**Reply by 2nOrderEDO** [reply]
Hello @ra3xdh, thanks for your reply.

I see and understand the limitations. Can't promise anything but I will keep an eye to see if there is a way to contribute manpower to the project.

Regards

**Comment by ra3xdh** [comment]
See #1045 for the proposal of the adding the simulation using the main menu.
